### ISSUE TITLE
Stepper and Stepper Item - Ensure implicit list markup is avoided - (2036570227)

### ISSUE THREAD

**Issue by dqateam** [Esri/calcite-design-system#7837]
### Violation: 
 Ensure implicit list markup is avoided 

![image](https://github.com/Esri/calcite-design-system/assets/5023024/dbe39068-671b-47d4-b08a-670f832300ad)

### WCAG Reference: 
- [1.3.1 Info and Relationships](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html) 

### Severity: 
 6 

### Media Type: 
 Lists 

--- 
### Areas for Remediation: 
- Title: Stepper and Stepper Item - Ensure implicit list markup is avoided - (2036570227) 
- Module: Stepper and Stepper Item 
**Issue**
There is content that appears as a list but does not use list markup. Examples include:
- Complete property
- Error property
- Active property
- Disabled property

**User Impact**
Screen reader users will have difficulty understanding that this content is a list.

**Code Reference** 
 ```
<div class="stepper-item-header-text"><span class="stepper-item-heading">Complete property</span><span class="stepper-item-description"></span></div>
(...)
<div class="stepper-item-header-text"><span class="stepper-item-heading">Error property</span><span class="stepper-item-description"></span></div>
(...)
<div class="stepper-item-header-text"><span class="stepper-item-heading">Disabled property</span><span class="stepper-item-description"></span></div>
 ``` 
**Suggestion**
Ensure implicit list markup is avoided. Lists must use ``` <ul> ```, ``` <ol> ```, ``` <li> ```, ``` <dl> ```, ``` <dt> ```, and ``` <dd> ``` elements as appropriate.

**Compliant Code Example** 
 ```
<ul>
<li><div class="stepper-item-header-text"><span class="stepper-item-heading">Complete property</span><span class="stepper-item-description"></span></div></li>
(...)
<li><div class="stepper-item-header-text"><span class="stepper-item-heading">Error property</span><span class="stepper-item-description"></span></div></li>
(...)
<li><div class="stepper-item-header-text"><span class="stepper-item-heading">Disabled property</span><span class="stepper-item-description"></span></div></li>
</ul>
 ```
 
--- 
### Additional Resources: 
- Report Source: Product accessibility evaluation conducted on the Calcite Design System in August, 2022. 

**monday.com sync:** #6365557217

**Comment by geospatialem** [comment]
It looks like the component's role is currently a `"region"`, do we want to explore a different role? Perhaps a `"tablist"` with individual `stepper-item`s as `"tab"s similar to this [tabs example](https://www.w3.org/WAI/ARIA/apg/patterns/tabs/examples/tabs-automatic/)?

https://github.com/Esri/calcite-design-system/blob/624c8b47e9f25b16be76711f7e9fa134125cc76b/packages/calcite-components/src/components/stepper/stepper.tsx#L173

**Comment by driskull** [comment]
Tablist role seems appropriate here. I think this component should be refactored to implement that role.

**Comment by geospatialem** [comment]
Discussed and confirmed the role change with @SkyeSeitz - we're `ready for dev`. ☄️

**Comment by geospatialem** [comment]
Related: https://github.com/Esri/calcite-design-system/issues/9147

**Comment by anveshmekala** [comment]
This issue is created in conjunction with https://github.com/Esri/calcite-design-system/issues/5604 ,https://github.com/Esri/calcite-design-system/issues/5602. Fix https://github.com/Esri/calcite-design-system/pull/7691 is installed post this issue and this can be closed.

Rationale behind using `region` role instead of `tablist`:

- `tablist` role for stepper means stepper-item's has a role of `tab`. Visually they are more alike but differ in keyboard functionalities & accessibility. 
- Tabs are often associated with closable property and stepper doesn't allow closing or removing.
- Users can move from one tab to other tab unrestricted where as `stepper-item's` can have restricted view depending on the status of current or previous tab
- slotted content in tab is accessible and currently stepper-item doesn't let users to tab through.
- Though `region` role do not capture the structure of stepper, combined with aria-label it provides context for AT users.

FWIW, refactoring stepper to use `<li/>` might help in providing more context as progress indicator. To accommodate `<li/> some of the keyboard navigation needs to be re-evaluated.

cc @geospatialem , @DitwanP

**Comment by DitwanP** [comment]
Closing out, see the comment above for why. 👆
